modyn/trainer_server/internal/dataset/key_sources/selector_key_source.py

```diff
@@ -105,16 +105,12 @@ def _connect_to_selector(self) -> SelectorStub:
             if grpc_connection_established(selector_channel):
                 return SelectorStub(selector_channel)
 
-            # no connection established
-            if attempt < max_retries:
-                logger.info(f"gRPC connection attempt {attempt} failed. Retrying in {retry_delay} seconds...")
-                time.sleep(retry_delay)
-                retry_delay *= 2  # Exponential backoff
-            else:
-                logger.error(f"Failed to establish gRPC connection after {max_retries} attempts.")
-                raise ConnectionError(
-                    f"Could not establish gRPC connection to selector at address {self._selector_address}."
-                )
```

Comment:
don't know how it went through pylint before.
But with the previous code, it's possible for this function to return inconsistent data types.

Reply:
I don't understand why it would be inconsistent (we either return or raise), but can you also adapt the online_dataset, then? we have the same problem there

Reply:
I don't understand it now. Why hasn't pylint complained about the other occurrence :( but I will make a consistent change